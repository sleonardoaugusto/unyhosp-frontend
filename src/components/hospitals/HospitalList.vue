<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(h, i) in hospitals"
      :key="i"
      class="card col-md-3"
      :to="{name: 'hospital-utis', params: {hospitalId: h.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ h.name }}</p>
      </div>
      <div class="row justify-content-between w-100">
        <router-link :to="{name: 'attendanceRegister', params: {hospitalId: h.id}}"
                     class="card__action type-light-green">Novo atendimento
        </router-link>
        <span class="card__action type-yellow">Editar</span>
      </div>
    </router-link>
  </div>
</template>

<script>
import hospitals from '@/mocks/hospitals'

export default {
  name: 'HospitalList',
  data: () => ({
    hospitals: []
  }),
  created() {
    this.hospitals = hospitals.data
  }
}
</script>

<style scoped>

</style>