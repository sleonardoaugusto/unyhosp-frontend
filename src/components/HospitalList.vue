<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(h, i) in hospitals"
      :key="i"
      class="card col-md-3"
      :to="{name: 'hospital-utis', params: {hospitalId: h.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ h.name }}</p>
      </div>
      <div class="row justify-content-between w-100">
        <span class="card__action type-blue">Novo atendimento</span>
        <span class="card__action type-yellow">Editar</span>
      </div>
    </router-link>
  </div>
</template>

<script>
import hospitals from '@/mocks/hospitals'

export default {
  name: 'HospitalList',
  data: () => ({
    hospitals: []
  }),
  created() {
    this.hospitals = hospitals.data
  }
}
</script>

<style scoped>

</style>