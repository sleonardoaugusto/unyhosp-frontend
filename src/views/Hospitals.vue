<template>
  <div class="col-md-12">
    <div class="row">
      <div class="col-md-12 d-flex justify-content-center">
        <h1 class="type-h3 type-blue-smoke">Hospitais</h1>
      </div>
    </div>
    <div v-if="showList" class="row">
      <router-link :to="{name: 'hospital-register'}" class="button button-green">
        <BaseIcon name="plus-circle"/>
        Novo Hospital
      </router-link>
      <hospital-list/>
    </div>
    <router-view/>
  </div>
</template>

<script>
import HospitalList from '@/components/hospitals/HospitalList'
import BaseIcon from '@/components/common/BaseIcon'

export default {
  name: 'home',
  components: {
    HospitalList,
    BaseIcon
  },
  computed: {
    showList() {
      return this.$route.name == 'hospitals'
    }
  }
}
</script>
