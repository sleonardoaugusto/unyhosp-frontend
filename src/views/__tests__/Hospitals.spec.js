import { shallowMount, createLocalVue } from '@vue/test-utils'
import Hospitals from '@/views/Hospitals'
import HospitalList from '@/components/hospitals/HospitalList'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('Hospitals.vue', () => {
  let wrapper

  wrapper = shallowMount(Hospitals, {
    localVue,
    router
  })

  it('renders component properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders child component', () => {
    expect(wrapper.find(HospitalList)).toBeTruthy()
  })
})