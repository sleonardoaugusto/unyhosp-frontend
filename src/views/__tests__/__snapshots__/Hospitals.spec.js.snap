// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hospitals.vue renders component properly 1`] = `
<div class="home col-md-12">
  <div class="row">
    <div class="col-md-12 d-flex justify-content-center">
      <h1>Hospitais</h1>
    </div>
  </div>
  <div class="row">
    <hospital-list-stub></hospital-list-stub>
  </div>
</div>
`;
